src/playwrightTestServer.ts

```diff
@@ -418,30 +417,16 @@ export class PlaywrightTestServer {
     return { connection, errors };
   }
 
-  private async _wireTestServer(testServer: TestServerConnection, reporter: reporterTypes.ReporterV2, token: vscodeTypes.CancellationToken) {
+  private _wireTestServer(testServer: TestServerConnection, reporter: reporterTypes.ReporterV2, token: vscodeTypes.CancellationToken, disposables: vscodeTypes.Disposable[]): void {
     const teleReceiver = new TeleReporterReceiver(reporter, {
       mergeProjects: true,
       mergeTestCases: true,
       resolvePath,
     });
-    return new Promise<void>(resolve => {
-      const disposables = [
-        testServer.onReport(async message => {
-          if (token.isCancellationRequested && message.method !== 'onEnd')
```

Comment:
I think this check is still useful, because events are coming asynchronously from the test runner, even after cancellation was requested.